src/Reflector/ClassReflector.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace BetterReflection\Reflector;
+
+use BetterReflection\Reflection\Symbol;
+use BetterReflection\Reflector\Generic as GenericReflector;
+use BetterReflection\SourceLocator\SourceLocator;
+
+class ClassReflector implements Reflector
+{
+    /**
+     * @var GenericReflector
+     */
+    private $reflector;
+
+    public function __construct(SourceLocator $sourceLocator)
+    {
+        $this->reflector = new GenericReflector($sourceLocator);
+    }
+
+    /**
+     * @param $className
+     * @return \BetterReflection\Reflection\ReflectionClass
+     */
+    public function reflect($className)
+    {
+        $symbol = new Symbol($className, Symbol::SYMBOL_CLASS);
```

Comment:
As suggested above, what is being attempted here is typing via internal state - not really suggested IMO